roachtest: liquibase failed

roachtest.liquibase [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3087613&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3087613&tab=artifacts#/liquibase) on master @ [ec36bc66f5147a8703719a9c890be12dd0c08945](https://github.com/cockroachdb/cockroach/commits/ec36bc66f5147a8703719a9c890be12dd0c08945):


```
		  | INFO: Custom SQL executed
		  | Jun 16, 2021 7:23:12 AM liquibase.changelog
		  | INFO: ChangeSet liquibase/harness/change/changelogs/sql.xml::sqlTest::kristyl ran successfully in 67ms
		  | Jun 16, 2021 7:23:12 AM liquibase.lockservice
		  | INFO: Successfully released change log lock
		  | Jun 16, 2021 7:23:12 AM java_util_logging_Logger$info$1 call
		  | INFO: Found 2 changeLogs for cockroachdb/20.2 in liquibase/harness/snapshot
		  | Error: COMMAND_PROBLEM: exit status 1
		  | (1) COMMAND_PROBLEM
		  | Wraps: (2) Node 1. Command with error:
		  |   | ``````
		  |   | cd /mnt/data1/liquibase-test-harness/ && mvn test -Dtest=LiquibaseHarnessSuiteTest -DdbName=cockroachdb -DdbVersion=20.2
		  |   | ``````
		  | Wraps: (3) exit status 1
		  | Error types: (1) errors.Cmd (2) *hintdetail.withDetail (3) *exec.ExitError
		  |
		  | stdout:
		  | <... some data truncated by circular buffer; go to artifacts for details ...>
		  | d SQL and DB snapshot(liquibase.harness.change.ChangeObjectTests): No signature of method: liquibase.command.core.SnapshotCommand.execute() is applicable for argument types: () values: [](..)
		  |   apply addNotNullConstraint against cockroachdb 20.2; verify generated SQL and DB snapshot(liquibase.harness.change.ChangeObjectTests): No signature of method: liquibase.command.core.SnapshotCommand.execute() is applicable for argument types: () values: [](..)
		  |   apply addUniqueConstraint against cockroachdb 20.2; verify generated SQL and DB snapshot(liquibase.harness.change.ChangeObjectTests): No signature of method: liquibase.command.core.SnapshotCommand.execute() is applicable for argument types: () values: [](..)
		  |   apply createIndex against cockroachdb 20.2; verify generated SQL and DB snapshot(liquibase.harness.change.ChangeObjectTests): No signature of method: liquibase.command.core.SnapshotCommand.execute() is applicable for argument types: () values: [](..)
		  |   apply createTableDataTypeText against cockroachdb 20.2; verify generated SQL and DB snapshot(liquibase.harness.change.ChangeObjectTests): No signature of method: liquibase.command.core.SnapshotCommand.execute() is applicable for argument types: () values: [](..)
		  |   apply createTableTimestamp against cockroachdb 20.2; verify generated SQL and DB snapshot(liquibase.harness.change.ChangeObjectTests): No signature of method: liquibase.command.core.SnapshotCommand.execute() is applicable for argument types: () values: [](..)
		  |   apply createView against cockroachdb 20.2; verify generated SQL and DB snapshot(liquibase.harness.change.ChangeObjectTests): No signature of method: liquibase.command.core.SnapshotCommand.execute() is applicable for argument types: () values: [](..)
		  |   apply dropColumn against cockroachdb 20.2; verify generated SQL and DB snapshot(liquibase.harness.change.ChangeObjectTests): No signature of method: liquibase.command.core.SnapshotCommand.execute() is applicable for argument types: () values: [](..)
		  |   apply dropDefaultValue against cockroachdb 20.2; verify generated SQL and DB snapshot(liquibase.harness.change.ChangeObjectTests): No signature of method: liquibase.command.core.SnapshotCommand.execute() is applicable for argument types: () values: [](..)
		  |   apply dropNotNullConstraint against cockroachdb 20.2; verify generated SQL and DB snapshot(liquibase.harness.change.ChangeObjectTests): No signature of method: liquibase.command.core.SnapshotCommand.execute() is applicable for argument types: () values: [](..)
		  |   apply sql against cockroachdb 20.2; verify generated SQL and DB snapshot(liquibase.harness.change.ChangeObjectTests): No signature of method: liquibase.command.core.SnapshotCommand.execute() is applicable for argument types: () values: [](..)
		  |
		  | Tests run: 32, Failures: 9, Errors: 14, Skipped: 5
		  |
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;34mINFO[m] [1;31mBUILD FAILURE[m
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;34mINFO[m] Total time:  43.660 s
		  | [[1;34mINFO[m] Finished at: 2021-06-16T07:23:20Z
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;31mERROR[m] Failed to execute goal [32morg.apache.maven.plugins:maven-surefire-plugin:2.12.4:test[m [1m(default-test)[m on project [36mliquibase-test-harness[m: [1;31mThere are test failures.[m
		  | [[1;31mERROR[m] [1;31m[m
		  | [[1;31mERROR[m] [1;31mPlease refer to /mnt/data1/liquibase-test-harness/target/surefire-reports for the individual test results.[m
		  | [[1;31mERROR[m] -> [1m[Help 1][m
		  | [[1;31mERROR[m] 
		  | [[1;31mERROR[m] To see the full stack trace of the errors, re-run Maven with the [1m-e[m switch.
		  | [[1;31mERROR[m] Re-run Maven using the [1m-X[m switch to enable full debug logging.
		  | [[1;31mERROR[m] 
		  | [[1;31mERROR[m] For more information about the errors and possible solutions, please read the following articles:
		  | [[1;31mERROR[m] [1m[Help 1][m http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
		Wraps: (4) exit status 20
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *main.withCommandDetails (4) *exec.ExitError
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh liquibase
```
</p>
</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*liquibase.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.liquibase [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3185657&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3185657&tab=artifacts#/liquibase) on master @ [31cdc0e4e0c5dda4cf67b53b45c36faf6a24eb33](https://github.com/cockroachdb/cockroach/commits/31cdc0e4e0c5dda4cf67b53b45c36faf6a24eb33):


```
		  | |                                             "defaultValue": "nextval\\('test_sequence'::STRING\\)\\!\\{liquibase.statement.DatabaseFunction\\}",
		  | |                                             "name": "sequence_referenced_column",
		  | |                                             "nullable": true,
		  | |                                             "type": {
		  | |                                               "typeName": "bigserial"
		  | |                                             }
		  | |                                           }
		  | |                                         }
		  | |                                       ]
		  | |                                     }
		  | |                                   }
		  | |                                 }
		  | java.lang.AssertionError: snapshot.objects.liquibase.structure.core.Column[0] Could not find match for element {"column":{"nullable":true,"defaultValue":"nextval\\('test_sequence'::STRING\\)\\!\\{liquibase.statement.DatabaseFunction\\}","name":"sequence_referenced_column","type":{"typeName":"bigserial"}}}
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:482)
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:464)
		  | 	at liquibase.harness.change.ChangeObjectTestHelper.snapshotMatchesSpecifiedStructure(ChangeObjectTestHelper.groovy:84)
		  | 	at liquibase.harness.change.ChangeObjectTests.apply #testInput.changeObject against #testInput.databaseName #testInput.version; verify generated SQL and DB snapshot(ChangeObjectTests.groovy:61)
		  |
		  | 	at liquibase.harness.change.ChangeObjectTests.apply #testInput.changeObject against #testInput.databaseName #testInput.version; verify generated SQL and DB snapshot(ChangeObjectTests.groovy:61)
		  | Caused by: java.lang.AssertionError: snapshot.objects.liquibase.structure.core.Column[0] Could not find match for element {"column":{"nullable":true,"defaultValue":"nextval\\('test_sequence'::STRING\\)\\!\\{liquibase.statement.DatabaseFunction\\}","name":"sequence_referenced_column","type":{"typeName":"bigserial"}}}
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:482)
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:464)
		  | 	at liquibase.harness.change.ChangeObjectTestHelper.snapshotMatchesSpecifiedStructure(ChangeObjectTestHelper.groovy:84)
		  | 	... 1 more
		  |
		  |
		  | Results :
		  |
		  | Failed tests:   apply addDefaultValueSequenceNext against cockroachdb 20.2; verify generated SQL and DB snapshot(liquibase.harness.change.ChangeObjectTests): Condition failed with Exception:(..)
		  |
		  | Tests run: 42, Failures: 1, Errors: 0, Skipped: 5
		  |
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;34mINFO[m] [1;31mBUILD FAILURE[m
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;34mINFO[m] Total time:  01:58 min
		  | [[1;34mINFO[m] Finished at: 2021-07-15T09:27:05Z
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;31mERROR[m] Failed to execute goal [32morg.apache.maven.plugins:maven-surefire-plugin:2.12.4:test[m [1m(default-test)[m on project [36mliquibase-test-harness[m: [1;31mThere are test failures.[m
		  | [[1;31mERROR[m] [1;31m[m
		  | [[1;31mERROR[m] [1;31mPlease refer to /mnt/data1/liquibase-test-harness/target/surefire-reports for the individual test results.[m
		  | [[1;31mERROR[m] -> [1m[Help 1][m
		  | [[1;31mERROR[m] 
		  | [[1;31mERROR[m] To see the full stack trace of the errors, re-run Maven with the [1m-e[m switch.
		  | [[1;31mERROR[m] Re-run Maven using the [1m-X[m switch to enable full debug logging.
		  | [[1;31mERROR[m] 
		  | [[1;31mERROR[m] For more information about the errors and possible solutions, please read the following articles:
		  | [[1;31mERROR[m] [1m[Help 1][m http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
		Wraps: (4) exit status 20
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *cluster.WithCommandDetails (4) *exec.ExitError
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh liquibase
```
</p>
</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*liquibase.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.liquibase [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3189962&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3189962&tab=artifacts#/liquibase) on master @ [4e2ee35288faf691d655d51e315a14fffab431ac](https://github.com/cockroachdb/cockroach/commits/4e2ee35288faf691d655d51e315a14fffab431ac):


```
		  | |                                             "defaultValue": "nextval\\('test_sequence'::STRING\\)\\!\\{liquibase.statement.DatabaseFunction\\}",
		  | |                                             "name": "sequence_referenced_column",
		  | |                                             "nullable": true,
		  | |                                             "type": {
		  | |                                               "typeName": "bigserial"
		  | |                                             }
		  | |                                           }
		  | |                                         }
		  | |                                       ]
		  | |                                     }
		  | |                                   }
		  | |                                 }
		  | java.lang.AssertionError: snapshot.objects.liquibase.structure.core.Column[0] Could not find match for element {"column":{"nullable":true,"defaultValue":"nextval\\('test_sequence'::STRING\\)\\!\\{liquibase.statement.DatabaseFunction\\}","name":"sequence_referenced_column","type":{"typeName":"bigserial"}}}
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:482)
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:464)
		  | 	at liquibase.harness.change.ChangeObjectTestHelper.snapshotMatchesSpecifiedStructure(ChangeObjectTestHelper.groovy:84)
		  | 	at liquibase.harness.change.ChangeObjectTests.apply #testInput.changeObject against #testInput.databaseName #testInput.version; verify generated SQL and DB snapshot(ChangeObjectTests.groovy:61)
		  |
		  | 	at liquibase.harness.change.ChangeObjectTests.apply #testInput.changeObject against #testInput.databaseName #testInput.version; verify generated SQL and DB snapshot(ChangeObjectTests.groovy:61)
		  | Caused by: java.lang.AssertionError: snapshot.objects.liquibase.structure.core.Column[0] Could not find match for element {"column":{"nullable":true,"defaultValue":"nextval\\('test_sequence'::STRING\\)\\!\\{liquibase.statement.DatabaseFunction\\}","name":"sequence_referenced_column","type":{"typeName":"bigserial"}}}
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:482)
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:464)
		  | 	at liquibase.harness.change.ChangeObjectTestHelper.snapshotMatchesSpecifiedStructure(ChangeObjectTestHelper.groovy:84)
		  | 	... 1 more
		  |
		  |
		  | Results :
		  |
		  | Failed tests:   apply addDefaultValueSequenceNext against cockroachdb 20.2; verify generated SQL and DB snapshot(liquibase.harness.change.ChangeObjectTests): Condition failed with Exception:(..)
		  |
		  | Tests run: 42, Failures: 1, Errors: 0, Skipped: 5
		  |
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;34mINFO[m] [1;31mBUILD FAILURE[m
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;34mINFO[m] Total time:  01:48 min
		  | [[1;34mINFO[m] Finished at: 2021-07-16T08:54:05Z
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;31mERROR[m] Failed to execute goal [32morg.apache.maven.plugins:maven-surefire-plugin:2.12.4:test[m [1m(default-test)[m on project [36mliquibase-test-harness[m: [1;31mThere are test failures.[m
		  | [[1;31mERROR[m] [1;31m[m
		  | [[1;31mERROR[m] [1;31mPlease refer to /mnt/data1/liquibase-test-harness/target/surefire-reports for the individual test results.[m
		  | [[1;31mERROR[m] -> [1m[Help 1][m
		  | [[1;31mERROR[m] 
		  | [[1;31mERROR[m] To see the full stack trace of the errors, re-run Maven with the [1m-e[m switch.
		  | [[1;31mERROR[m] Re-run Maven using the [1m-X[m switch to enable full debug logging.
		  | [[1;31mERROR[m] 
		  | [[1;31mERROR[m] For more information about the errors and possible solutions, please read the following articles:
		  | [[1;31mERROR[m] [1m[Help 1][m http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
		Wraps: (4) exit status 20
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *cluster.WithCommandDetails (4) *exec.ExitError
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh liquibase
```
</p>
</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*liquibase.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.liquibase [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3194875&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3194875&tab=artifacts#/liquibase) on master @ [efb4fd2b406cd47b505b4f1a1d5700fc79841469](https://github.com/cockroachdb/cockroach/commits/efb4fd2b406cd47b505b4f1a1d5700fc79841469):


```
		  | |                                             "defaultValue": "nextval\\('test_sequence'::STRING\\)\\!\\{liquibase.statement.DatabaseFunction\\}",
		  | |                                             "name": "sequence_referenced_column",
		  | |                                             "nullable": true,
		  | |                                             "type": {
		  | |                                               "typeName": "bigserial"
		  | |                                             }
		  | |                                           }
		  | |                                         }
		  | |                                       ]
		  | |                                     }
		  | |                                   }
		  | |                                 }
		  | java.lang.AssertionError: snapshot.objects.liquibase.structure.core.Column[0] Could not find match for element {"column":{"nullable":true,"defaultValue":"nextval\\('test_sequence'::STRING\\)\\!\\{liquibase.statement.DatabaseFunction\\}","name":"sequence_referenced_column","type":{"typeName":"bigserial"}}}
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:482)
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:464)
		  | 	at liquibase.harness.change.ChangeObjectTestHelper.snapshotMatchesSpecifiedStructure(ChangeObjectTestHelper.groovy:84)
		  | 	at liquibase.harness.change.ChangeObjectTests.apply #testInput.changeObject against #testInput.databaseName #testInput.version; verify generated SQL and DB snapshot(ChangeObjectTests.groovy:61)
		  |
		  | 	at liquibase.harness.change.ChangeObjectTests.apply #testInput.changeObject against #testInput.databaseName #testInput.version; verify generated SQL and DB snapshot(ChangeObjectTests.groovy:61)
		  | Caused by: java.lang.AssertionError: snapshot.objects.liquibase.structure.core.Column[0] Could not find match for element {"column":{"nullable":true,"defaultValue":"nextval\\('test_sequence'::STRING\\)\\!\\{liquibase.statement.DatabaseFunction\\}","name":"sequence_referenced_column","type":{"typeName":"bigserial"}}}
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:482)
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:464)
		  | 	at liquibase.harness.change.ChangeObjectTestHelper.snapshotMatchesSpecifiedStructure(ChangeObjectTestHelper.groovy:84)
		  | 	... 1 more
		  |
		  |
		  | Results :
		  |
		  | Failed tests:   apply addDefaultValueSequenceNext against cockroachdb 20.2; verify generated SQL and DB snapshot(liquibase.harness.change.ChangeObjectTests): Condition failed with Exception:(..)
		  |
		  | Tests run: 42, Failures: 1, Errors: 0, Skipped: 5
		  |
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;34mINFO[m] [1;31mBUILD FAILURE[m
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;34mINFO[m] Total time:  01:55 min
		  | [[1;34mINFO[m] Finished at: 2021-07-19T17:02:44Z
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;31mERROR[m] Failed to execute goal [32morg.apache.maven.plugins:maven-surefire-plugin:2.12.4:test[m [1m(default-test)[m on project [36mliquibase-test-harness[m: [1;31mThere are test failures.[m
		  | [[1;31mERROR[m] [1;31m[m
		  | [[1;31mERROR[m] [1;31mPlease refer to /mnt/data1/liquibase-test-harness/target/surefire-reports for the individual test results.[m
		  | [[1;31mERROR[m] -> [1m[Help 1][m
		  | [[1;31mERROR[m] 
		  | [[1;31mERROR[m] To see the full stack trace of the errors, re-run Maven with the [1m-e[m switch.
		  | [[1;31mERROR[m] Re-run Maven using the [1m-X[m switch to enable full debug logging.
		  | [[1;31mERROR[m] 
		  | [[1;31mERROR[m] For more information about the errors and possible solutions, please read the following articles:
		  | [[1;31mERROR[m] [1m[Help 1][m http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
		Wraps: (4) exit status 20
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *cluster.WithCommandDetails (4) *exec.ExitError
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh liquibase
```
</p>
</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*liquibase.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.liquibase [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3194041&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3194041&tab=artifacts#/liquibase) on master @ [fa1da4ac2872533007e4ed4c70127f85040ecb25](https://github.com/cockroachdb/cockroach/commits/fa1da4ac2872533007e4ed4c70127f85040ecb25):


```
		  | |                                             "defaultValue": "nextval\\('test_sequence'::STRING\\)\\!\\{liquibase.statement.DatabaseFunction\\}",
		  | |                                             "name": "sequence_referenced_column",
		  | |                                             "nullable": true,
		  | |                                             "type": {
		  | |                                               "typeName": "bigserial"
		  | |                                             }
		  | |                                           }
		  | |                                         }
		  | |                                       ]
		  | |                                     }
		  | |                                   }
		  | |                                 }
		  | java.lang.AssertionError: snapshot.objects.liquibase.structure.core.Column[0] Could not find match for element {"column":{"nullable":true,"defaultValue":"nextval\\('test_sequence'::STRING\\)\\!\\{liquibase.statement.DatabaseFunction\\}","name":"sequence_referenced_column","type":{"typeName":"bigserial"}}}
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:482)
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:464)
		  | 	at liquibase.harness.change.ChangeObjectTestHelper.snapshotMatchesSpecifiedStructure(ChangeObjectTestHelper.groovy:84)
		  | 	at liquibase.harness.change.ChangeObjectTests.apply #testInput.changeObject against #testInput.databaseName #testInput.version; verify generated SQL and DB snapshot(ChangeObjectTests.groovy:61)
		  |
		  | 	at liquibase.harness.change.ChangeObjectTests.apply #testInput.changeObject against #testInput.databaseName #testInput.version; verify generated SQL and DB snapshot(ChangeObjectTests.groovy:61)
		  | Caused by: java.lang.AssertionError: snapshot.objects.liquibase.structure.core.Column[0] Could not find match for element {"column":{"nullable":true,"defaultValue":"nextval\\('test_sequence'::STRING\\)\\!\\{liquibase.statement.DatabaseFunction\\}","name":"sequence_referenced_column","type":{"typeName":"bigserial"}}}
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:482)
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:464)
		  | 	at liquibase.harness.change.ChangeObjectTestHelper.snapshotMatchesSpecifiedStructure(ChangeObjectTestHelper.groovy:84)
		  | 	... 1 more
		  |
		  |
		  | Results :
		  |
		  | Failed tests:   apply addDefaultValueSequenceNext against cockroachdb 20.2; verify generated SQL and DB snapshot(liquibase.harness.change.ChangeObjectTests): Condition failed with Exception:(..)
		  |
		  | Tests run: 42, Failures: 1, Errors: 0, Skipped: 5
		  |
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;34mINFO[m] [1;31mBUILD FAILURE[m
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;34mINFO[m] Total time:  01:52 min
		  | [[1;34mINFO[m] Finished at: 2021-07-19T17:36:56Z
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;31mERROR[m] Failed to execute goal [32morg.apache.maven.plugins:maven-surefire-plugin:2.12.4:test[m [1m(default-test)[m on project [36mliquibase-test-harness[m: [1;31mThere are test failures.[m
		  | [[1;31mERROR[m] [1;31m[m
		  | [[1;31mERROR[m] [1;31mPlease refer to /mnt/data1/liquibase-test-harness/target/surefire-reports for the individual test results.[m
		  | [[1;31mERROR[m] -> [1m[Help 1][m
		  | [[1;31mERROR[m] 
		  | [[1;31mERROR[m] To see the full stack trace of the errors, re-run Maven with the [1m-e[m switch.
		  | [[1;31mERROR[m] Re-run Maven using the [1m-X[m switch to enable full debug logging.
		  | [[1;31mERROR[m] 
		  | [[1;31mERROR[m] For more information about the errors and possible solutions, please read the following articles:
		  | [[1;31mERROR[m] [1m[Help 1][m http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
		Wraps: (4) exit status 20
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *cluster.WithCommandDetails (4) *exec.ExitError
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh liquibase
```
</p>
</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*liquibase.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.liquibase [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3193792&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3193792&tab=artifacts#/liquibase) on master @ [5a5b3dc446fcfc2d3e28b6775ae9bb1a63376210](https://github.com/cockroachdb/cockroach/commits/5a5b3dc446fcfc2d3e28b6775ae9bb1a63376210):


```
		  | |                                             "defaultValue": "nextval\\('test_sequence'::STRING\\)\\!\\{liquibase.statement.DatabaseFunction\\}",
		  | |                                             "name": "sequence_referenced_column",
		  | |                                             "nullable": true,
		  | |                                             "type": {
		  | |                                               "typeName": "bigserial"
		  | |                                             }
		  | |                                           }
		  | |                                         }
		  | |                                       ]
		  | |                                     }
		  | |                                   }
		  | |                                 }
		  | java.lang.AssertionError: snapshot.objects.liquibase.structure.core.Column[0] Could not find match for element {"column":{"nullable":true,"defaultValue":"nextval\\('test_sequence'::STRING\\)\\!\\{liquibase.statement.DatabaseFunction\\}","name":"sequence_referenced_column","type":{"typeName":"bigserial"}}}
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:482)
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:464)
		  | 	at liquibase.harness.change.ChangeObjectTestHelper.snapshotMatchesSpecifiedStructure(ChangeObjectTestHelper.groovy:84)
		  | 	at liquibase.harness.change.ChangeObjectTests.apply #testInput.changeObject against #testInput.databaseName #testInput.version; verify generated SQL and DB snapshot(ChangeObjectTests.groovy:61)
		  |
		  | 	at liquibase.harness.change.ChangeObjectTests.apply #testInput.changeObject against #testInput.databaseName #testInput.version; verify generated SQL and DB snapshot(ChangeObjectTests.groovy:61)
		  | Caused by: java.lang.AssertionError: snapshot.objects.liquibase.structure.core.Column[0] Could not find match for element {"column":{"nullable":true,"defaultValue":"nextval\\('test_sequence'::STRING\\)\\!\\{liquibase.statement.DatabaseFunction\\}","name":"sequence_referenced_column","type":{"typeName":"bigserial"}}}
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:482)
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:464)
		  | 	at liquibase.harness.change.ChangeObjectTestHelper.snapshotMatchesSpecifiedStructure(ChangeObjectTestHelper.groovy:84)
		  | 	... 1 more
		  |
		  |
		  | Results :
		  |
		  | Failed tests:   apply addDefaultValueSequenceNext against cockroachdb 20.2; verify generated SQL and DB snapshot(liquibase.harness.change.ChangeObjectTests): Condition failed with Exception:(..)
		  |
		  | Tests run: 42, Failures: 1, Errors: 0, Skipped: 5
		  |
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;34mINFO[m] [1;31mBUILD FAILURE[m
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;34mINFO[m] Total time:  01:56 min
		  | [[1;34mINFO[m] Finished at: 2021-07-19T17:37:07Z
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;31mERROR[m] Failed to execute goal [32morg.apache.maven.plugins:maven-surefire-plugin:2.12.4:test[m [1m(default-test)[m on project [36mliquibase-test-harness[m: [1;31mThere are test failures.[m
		  | [[1;31mERROR[m] [1;31m[m
		  | [[1;31mERROR[m] [1;31mPlease refer to /mnt/data1/liquibase-test-harness/target/surefire-reports for the individual test results.[m
		  | [[1;31mERROR[m] -> [1m[Help 1][m
		  | [[1;31mERROR[m] 
		  | [[1;31mERROR[m] To see the full stack trace of the errors, re-run Maven with the [1m-e[m switch.
		  | [[1;31mERROR[m] Re-run Maven using the [1m-X[m switch to enable full debug logging.
		  | [[1;31mERROR[m] 
		  | [[1;31mERROR[m] For more information about the errors and possible solutions, please read the following articles:
		  | [[1;31mERROR[m] [1m[Help 1][m http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
		Wraps: (4) exit status 20
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *cluster.WithCommandDetails (4) *exec.ExitError
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh liquibase
```
</p>
</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*liquibase.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.liquibase [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3199213&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3199213&tab=artifacts#/liquibase) on master @ [5fd1dda1cde0a91643d39209c6458dcefcbfc180](https://github.com/cockroachdb/cockroach/commits/5fd1dda1cde0a91643d39209c6458dcefcbfc180):


```
		  | |                                             "defaultValue": "nextval\\('test_sequence'::STRING\\)\\!\\{liquibase.statement.DatabaseFunction\\}",
		  | |                                             "name": "sequence_referenced_column",
		  | |                                             "nullable": true,
		  | |                                             "type": {
		  | |                                               "typeName": "bigserial"
		  | |                                             }
		  | |                                           }
		  | |                                         }
		  | |                                       ]
		  | |                                     }
		  | |                                   }
		  | |                                 }
		  | java.lang.AssertionError: snapshot.objects.liquibase.structure.core.Column[0] Could not find match for element {"column":{"nullable":true,"defaultValue":"nextval\\('test_sequence'::STRING\\)\\!\\{liquibase.statement.DatabaseFunction\\}","name":"sequence_referenced_column","type":{"typeName":"bigserial"}}}
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:482)
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:464)
		  | 	at liquibase.harness.change.ChangeObjectTestHelper.snapshotMatchesSpecifiedStructure(ChangeObjectTestHelper.groovy:84)
		  | 	at liquibase.harness.change.ChangeObjectTests.apply #testInput.changeObject against #testInput.databaseName #testInput.version; verify generated SQL and DB snapshot(ChangeObjectTests.groovy:61)
		  |
		  | 	at liquibase.harness.change.ChangeObjectTests.apply #testInput.changeObject against #testInput.databaseName #testInput.version; verify generated SQL and DB snapshot(ChangeObjectTests.groovy:61)
		  | Caused by: java.lang.AssertionError: snapshot.objects.liquibase.structure.core.Column[0] Could not find match for element {"column":{"nullable":true,"defaultValue":"nextval\\('test_sequence'::STRING\\)\\!\\{liquibase.statement.DatabaseFunction\\}","name":"sequence_referenced_column","type":{"typeName":"bigserial"}}}
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:482)
		  | 	at org.skyscreamer.jsonassert.JSONAssert.assertEquals(JSONAssert.java:464)
		  | 	at liquibase.harness.change.ChangeObjectTestHelper.snapshotMatchesSpecifiedStructure(ChangeObjectTestHelper.groovy:84)
		  | 	... 1 more
		  |
		  |
		  | Results :
		  |
		  | Failed tests:   apply addDefaultValueSequenceNext against cockroachdb 20.2; verify generated SQL and DB snapshot(liquibase.harness.change.ChangeObjectTests): Condition failed with Exception:(..)
		  |
		  | Tests run: 42, Failures: 1, Errors: 0, Skipped: 5
		  |
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;34mINFO[m] [1;31mBUILD FAILURE[m
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;34mINFO[m] Total time:  01:54 min
		  | [[1;34mINFO[m] Finished at: 2021-07-20T08:44:31Z
		  | [[1;34mINFO[m] [1m------------------------------------------------------------------------[m
		  | [[1;31mERROR[m] Failed to execute goal [32morg.apache.maven.plugins:maven-surefire-plugin:2.12.4:test[m [1m(default-test)[m on project [36mliquibase-test-harness[m: [1;31mThere are test failures.[m
		  | [[1;31mERROR[m] [1;31m[m
		  | [[1;31mERROR[m] [1;31mPlease refer to /mnt/data1/liquibase-test-harness/target/surefire-reports for the individual test results.[m
		  | [[1;31mERROR[m] -> [1m[Help 1][m
		  | [[1;31mERROR[m] 
		  | [[1;31mERROR[m] To see the full stack trace of the errors, re-run Maven with the [1m-e[m switch.
		  | [[1;31mERROR[m] Re-run Maven using the [1m-X[m switch to enable full debug logging.
		  | [[1;31mERROR[m] 
		  | [[1;31mERROR[m] For more information about the errors and possible solutions, please read the following articles:
		  | [[1;31mERROR[m] [1m[Help 1][m http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
		Wraps: (4) exit status 20
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *cluster.WithCommandDetails (4) *exec.ExitError
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh liquibase
```
</p>
</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*liquibase.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/cmd/roachtest/tests/BUILD.bazel
+++ b/pkg/cmd/roachtest/tests/BUILD.bazel
@@ -61,6 +61,7 @@ go_library(
         "libpq.go",
         "libpq_blocklist.go",
         "liquibase.go",
+        "liquibase_blocklist.go",
         "many_splits.go",
         "mixed_version_decommission.go",
         "mixed_version_jobs.go",

--- a/pkg/cmd/roachtest/tests/liquibase.go
+++ b/pkg/cmd/roachtest/tests/liquibase.go
@@ -99,14 +99,34 @@ func registerLiquibase(r registry.Registry) {
 		}
 
 		t.Status("running liquibase test harness")
-		// All tests are expected to pass, so this should not error.
-		// The dbVersion is set to 20.2 since that causes all known passing tests
-		// to be run.
-		if err = c.RunE(ctx, node,
-			`cd /mnt/data1/liquibase-test-harness/ && mvn test -Dtest=LiquibaseHarnessSuiteTest -DdbName=cockroachdb -DdbVersion=20.2`,
-		); err != nil {
-			t.Fatal(err)
+		blocklistName, expectedFailures, _, ignoreList := liquibaseBlocklists.getLists(version)
+		if expectedFailures == nil {
+			t.Fatalf("No hibernate blocklist defined for cockroach version %s", version)
 		}
+
+		const (
+			repoDir     = "/mnt/data1/liquibase-test-harness"
+			resultsPath = repoDir + "/target/surefire-reports/TEST-liquibase.harness.LiquibaseHarnessSuiteTest.xml"
+		)
+
+		cmd := fmt.Sprintf("cd /mnt/data1/liquibase-test-harness/ && "+
+			"mvn surefire-report:report-only test -Dtest=LiquibaseHarnessSuiteTest "+
+			"-DdbName=cockroachdb -DdbVersion=20.2 -DoutputDirectory=%s", repoDir)
+
+		err = c.RunE(ctx, node, cmd)
+		if err != nil {
+			t.L().Printf("error whilst running tests (may be expected): %#v", err)
+		}
+
+		parseAndSummarizeJavaORMTestsResults(
+			ctx, t, c, node, "liquibase" /* ormName */, []byte(resultsPath),
+			blocklistName,
+			expectedFailures,
+			ignoreList,
+			version,
+			supportedLiquibaseHarnessCommit,
+		)
+
 	}
 
 	r.Add(registry.TestSpec{

--- a/pkg/cmd/roachtest/tests/liquibase_blocklist.go
+++ b/pkg/cmd/roachtest/tests/liquibase_blocklist.go
@@ -1,0 +1,31 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tests
+
+var liquibaseBlocklists = blocklistsForVersion{
+	{"v20.2", "liquibaseBlocklist20_2", liquibaseBlocklist20_2, "liquibaseIgnorelist20_2", liquibaseIgnorelist20_2},
+	{"v21.1", "liquibaseBlocklist21_1", liquibaseBlocklist21_1, "liquibaseIgnorelist21_1", liquibaseIgnorelist21_1},
+	{"v21.2", "liquibaseBlocklist21_2", liquibaseBlocklist21_2, "liquibaseIgnorelist21_2", liquibaseIgnorelist21_2},
+}
+
+var liquibaseBlocklist21_2 = blocklist{
+	"liquibase.harness.change.ChangeObjectTests.apply addDefaultValueSequenceNext against cockroachdb 20.2; verify generated SQL and DB snapshot": "",
+}
+
+var liquibaseBlocklist21_1 = liquibaseBlocklist20_2
+
+var liquibaseBlocklist20_2 = blocklist{}
+
+var liquibaseIgnorelist21_2 = liquibaseIgnorelist21_1
+
+var liquibaseIgnorelist21_1 = liquibaseIgnorelist20_2
+
+var liquibaseIgnorelist20_2 = blocklist{}

